### DISCUSSION TITLE
Guards to check string input's length

### DISCUSSION THREAD

**Discussion by rustam-r-hajiyev** [ardalis/GuardClauses#138] [Ideas]
Hello,

While using the library I didn't find guards that would help to ensure that string inputs are shorter than MAX and/or longer than MIN. It seems to me that these checks are quite common and it would be nice to have them in the library.

I already created a fork where I added 3 guards: MinLength, MaxLength and MinAndMaxLength: https://github.com/rustamhajiyev/GuardClauses

If you like these guards, I'd be happy to create a PR and contribute.

**Comment by chadboettcher** [comment]
I just created an extension in my project based in your code.  I only need to check for max length so I added two extensions.  The second just combines the NullOrEmpty with MaxLength because I normally want to check those both at the same time.  I'd love to see this get included in the main library.

```
public static class GuardExtensions
{
	public static string MaxLength(this IGuardClause guardClause, string input, string parameterName, uint maxLength, string message = null)
	{
		if (input?.Length > maxLength)
		{
			throw new ArgumentException(message ?? $"Required input {parameterName} cannot be longer than {maxLength} character(s).");
		}

		return input;
	}

	public static string NullOrEmptyOrMaxLength(this IGuardClause guardClause, string input, string parameterName, uint maxLength, string message = null)
	{
		Guard.Against.NullOrEmpty(input, parameterName, message);
		Guard.Against.MaxLength(input, parameterName, maxLength, message);

		return input;
	}
}
```
